nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/GetTCP.java

```diff
@@ -0,0 +1,668 @@
+        SocketRecveiverThread socketRecveiverThread;
+        socketRecveiverThread = new SocketRecveiverThread(hostAndPort[0],
+                Integer.parseInt(hostAndPort[1]), backupServer, keepAlive, rcvBufferSize, connectionTimeout, connectionRetryCount);
+        return socketRecveiverThread;
+    }
+
+    @OnStopped
+    public void tearDown() throws ProcessException {
+        try {
+            for (Map.Entry<SocketRecveiverThread, Future> socketAndFuture : socketToFuture.entrySet()) {
+                socketAndFuture.getKey().stopProcessing();
+                socketAndFuture.getValue().cancel(true);
+
+            }
+            if (null != executorService) {
+                executorService.shutdown();
+                if (!executorService.awaitTermination(5, TimeUnit.SECONDS)) {
+                    executorService.shutdownNow();
+
+                    if (!executorService.awaitTermination(5, TimeUnit.SECONDS))
+                        log.error("Executor service for receiver thread did not terminate");
+                }
+            }
+        } catch (final Exception e) {
+            throw new ProcessException(e);
+        }
+    }
+
+    @Override
+    public void onTrigger(ProcessContext context, ProcessSession session) throws ProcessException {
+
+        try {
+            final StringBuilder messages = new StringBuilder();
+            if (socketMessagesReceived.size() >= batchSize) {
+                for (int i = 0; i < batchSize; i++) {
+                    messages.append(socketMessagesReceived.poll(100, TimeUnit.MILLISECONDS));
+                }
+            } else {
+                messages.append(socketMessagesReceived.poll(100, TimeUnit.MILLISECONDS));
+            }
+            if (0 == messages.length()) {
+                return;
+            }
+            FlowFile flowFile = session.create();
+
+            flowFile = session.write(flowFile, out -> out.write(messages.toString().getBytes()));
+            //need to at least put the list of hosts and ports, would be nice to know the exact
+            //host the message came frome -- to do that we need to do one of two things
+            //
+            // 1. Have a message queue per server
+            // 2. Add something to the message that then needs to be parsed.
+            //
+            // Both of these are not ideal, for now just show the whole list and someone can still use a
+            // ROA processor and see if the attribute contains a host.
+            flowFile = session.putAttribute(flowFile, "ServerList", originalServerAddressList);
+            //add any dynamic properties
+            if (0 < dynamicAttributes.size()) {
+                flowFile = session.putAllAttributes(flowFile, dynamicAttributes);
+            }
+            session.transfer(flowFile, REL_SUCCESS);
+
+        } catch (InterruptedException exception) {
+            throw new ProcessException(exception);
```

Comment:
The InterruptedException is usually the result if abnormal shutdown and not an application error. All we should to do here  is log (optional) and reset the interrupt flag for the current thread so other component in the call stack can react:
```
} catch (InterruptedException exception) {
            this.getLogger().warn("Current thread is interrupted");
            Thread.currentThread().interrupt();
}
```